### PR TITLE
add local registry for tinkerbell to host container images and OS images

### PR DESCRIPTION
**Pull request by gwesterfieldjr**
add local registry for tinkerbell to host container images and OS images

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jaxesn — COMMENTED** [review]
Is this temporary for beta?

**Comment by gwesterfieldjr** [comment]
> Is this temporary for beta?

Hmm, technically yes since in GA the tink stack will be in the bootstrap.

File: projects/tinkerbell/sandbox-local-registry/.env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+TINK_CLI_IMAGE="public.ecr.aws/l0g8r8j6/tinkerbell/tink/tink-cli:v0.6.0-eks-a-v0.0.0-dev-build.1536"
+TINK_SERVER_IMAGE="public.ecr.aws/l0g8r8j6/tinkerbell/tink/tink-server:v0.6.0-eks-a-v0.0.0-dev-build.1536"
+BOOTS_SERVER_IMAGE="public.ecr.aws/l0g8r8j6/tinkerbell/boots:v0.6.0-eks-a-v0.0.0-dev-build.1536"
+HEGEL_SERVER_IMAGE="public.ecr.aws/l0g8r8j6/tinkerbell/hegel:v0.6.0-eks-a-v0.0.0-dev-build.1536"
+PBNJ_IMAGE_URL="public.ecr.aws/l0g8r8j6/tinkerbell/pbnj:7275abd396e5fce9b1358e18fe6d935f6cb341af-eks-a-v0.0.0-dev-build.1536"
```
Comment (abhinavmpandey08) [inline projects/tinkerbell/sandbox-local-registry/.env:5]: can we point these to the later builds? I see the latest ones are tagged `v0.6.0-eks-a-v0.0.0-dev-build.2151`

File: projects/tinkerbell/sandbox-local-registry/tls/trust.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+systemctl restart docker
+
+# login into the docker registry.
+docker login "$TINKERBELL_HOST_IP" --username admin --password-stdin <<<'Admin1234'
+if id -u vagrant >/dev/null 2>&1; then
+	su vagrant -c "docker login \"$TINKERBELL_HOST_IP\" --username admin --password-stdin" <<<'Admin1234'
+fi
+
+# ensure everything is up after docker restart.
+docker-compose up --detach
```
Comment (abhinavmpandey08) [inline projects/tinkerbell/sandbox-local-registry/tls/trust.sh:22]: optional, but i dont think this script gets used anywhere

File: projects/tinkerbell/sandbox-local-registry/registry/registry_images.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+public.ecr.aws/l0g8r8j6/tinkerbell/tink/tink-worker:v0.6.0-eks-a-v0.0.0-dev-build.1627 tink-worker:latest
+public.ecr.aws/l0g8r8j6/tinkerbell/hub/image2disk:6c0f0d437bde2c836d90b000312c8b25fa1b65e1-eks-a-v0.0.0-dev-build.1536 image2disk:v1.0.0
+public.ecr.aws/l0g8r8j6/tinkerbell/hub/cexec:6c0f0d437bde2c836d90b000312c8b25fa1b65e1-eks-a-v0.0.0-dev-build.1536 cexec:v1.0.0
+public.ecr.aws/l0g8r8j6/tinkerbell/hub/writefile:6c0f0d437bde2c836d90b000312c8b25fa1b65e1-eks-a-v0.0.0-dev-build.1536 writefile:v1.0.0
+public.ecr.aws/l0g8r8j6/tinkerbell/hub/kexec:6c0f0d437bde2c836d90b000312c8b25fa1b65e1-eks-a-v0.0.0-dev-build.1536 kexec:v1.0.0
+public.ecr.aws/l0g8r8j6/tinkerbell/hub/oci2disk:6c0f0d437bde2c836d90b000312c8b25fa1b65e1-eks-a-v0.0.0-dev-build.1536 oci2disk:v1.0.0
```
Comment (abhinavmpandey08) [inline projects/tinkerbell/sandbox-local-registry/registry/registry_images.txt:6]: can you also update these to the newer tags?

File: projects/tinkerbell/sandbox-local-registry/registry/os_images.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+https://projectbuildpipeline-857-pipelineoutputartifactsb-10ajmk30khe3f.s3.us-west-2.amazonaws.com/projects/tinkerbell/hook/latest/54bae38625828b69259eccb3357e13c269437e28/vmlinuz-x86_64 misc/osie/current/vmlinuz-x86_64
+https://projectbuildpipeline-857-pipelineoutputartifactsb-10ajmk30khe3f.s3.us-west-2.amazonaws.com/projects/tinkerbell/hook/latest/54bae38625828b69259eccb3357e13c269437e28/initramfs-x86_64 misc/osie/current/initramfs-x86_64
```
Comment (abhinavmpandey08) [inline projects/tinkerbell/sandbox-local-registry/registry/os_images.txt:2]: I think we want to pull these from the `dev-release-prod-pdx` bucket
```suggestion
https://dev-release-prod-pdx.s3.us-west-2.amazonaws.com/artifacts/v0.0.0-dev-build.2036/hook/54bae38625828b69259eccb3357e13c269437e28/vmlinuz-x86_64
https://dev-release-prod-pdx.s3.us-west-2.amazonaws.com/artifacts/v0.0.0-dev-build.2036/hook/54bae38625828b69259eccb3357e13c269437e28/initramfs-x86_64
```

File: projects/tinkerbell/sandbox-local-registry/.env (hunk 1 above)
Comment (gwesterfieldjr) [reply projects/tinkerbell/sandbox-local-registry/.env:5]: what about pbnj? it has a different tag?

File: projects/tinkerbell/sandbox-local-registry/registry/registry_images.txt (hunk 3 above)
Comment (gwesterfieldjr) [reply projects/tinkerbell/sandbox-local-registry/registry/registry_images.txt:6]: Did the commit id change as well or just the build number at the end of the tag?

Comment (abhinavmpandey08) [reply projects/tinkerbell/sandbox-local-registry/registry/registry_images.txt:6]: can you check ECR for the exact tags? Here's one for kexec for reference https://gallery.ecr.aws/l0g8r8j6/tinkerbell/hub/kexec

**Review by abhinavmpandey08 — APPROVED** [review]
/lgtm

**Pull request merged**